cmd/kube-scheduler/app/testing/testserver.go

```diff
@@ -120,7 +121,16 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 
 	errCh := make(chan error)
 	go func(stopCh <-chan struct{}) {
-		if err := app.Run(config.Complete(), stopCh); err != nil {
+		if config.InsecureServing != nil {
```

Comment:
This is introducing repeated code. What about moving the logic to `app.Run`?

Reply:
Makes sense. Done!